src/dispatch/messaging/email/utils.py

```diff
@@ -45,19 +46,26 @@ def get_template(message_type: MessageType):
         ),
     }
 
-    template_path, description = template_map.get(message_type, (None, None))
+    template_key, description = template_map.get(message_type, (None, None))
 
-    if not template_path:
+    if not template_key:
         raise Exception(f"Unable to determine template. MessageType: {message_type}")
 
-    return env.get_template(os.path.join("templates", template_path)), description
+    try:
+        template_path = os.path.join("templates", "project_id", f"{project_id}", template_key)
+        template = env.get_template(template_path)
+    except jinja2.exceptions.TemplateNotFound:
+        template_path = os.path.join("templates", template_key)
```

Comment:
We may want to add log a warning using `log.warn()`, if the template is not found and the default is used.

Reply:
But since this feature was not available before and nobody really requested it, sounds like most of the times the warning will be issued :)

do you think debug log will be more appropriate here? which will show what was the template path resolved?

Reply:
I added the debug for now, here is the example of how I tested it:

```
dispatch-web-1  | DEBUG:Resolved template path: templates/project_id/6/notification.mjml:/usr/local/lib/python3.11/site-packages/dispatch/messaging/email/utils.py:get_template:60
```